jib-core/src/main/java/com/google/cloud/tools/jib/image/json/ImageToJsonTranslator.java

```diff
@@ -59,26 +60,34 @@
   @VisibleForTesting
   @Nullable
   static Map<String, Map<?, ?>> portListToMap(@Nullable List<Port> exposedPorts) {
-    if (exposedPorts == null) {
-      return null;
-    }
+    return listToMap(exposedPorts, port -> port.getPort() + "/" + port.getProtocol());
+  }
 
-    return listToMap(
-        exposedPorts
-            .stream()
-            .map(port -> port.getPort() + "/" + port.getProtocol())
-            .collect(Collectors.toList()));
+  /**
+   * Converts a list of {@link AbsoluteUnixPath}s to the corresponding container config format for
+   * volumes (e.g. {@code AbsoluteUnixPath().get("/var/log/my-app-logs")} -> {@code
+   * {"/var/log/my-app-logs":{}}}).
+   *
+   * @param volumes the list of {@link AbsoluteUnixPath}s to translate, or {@code null}
+   * @return a sorted map with the string representation of the ports as keys and empty maps as
+   *     values, or {@code null} if {@code exposedPorts} is {@code null}
+   */
+  @VisibleForTesting
+  @Nullable
+  static Map<String, Map<?, ?>> volumesListToMap(@Nullable List<AbsoluteUnixPath> volumes) {
+    return listToMap(volumes, AbsoluteUnixPath::toString);
   }
 
   @Nullable
-  static Map<String, Map<?, ?>> listToMap(@Nullable List<String> list) {
+  static <E> Map<String, Map<?, ?>> listToMap(
+      @Nullable List<E> list, Function<E, String> elementMapper) {
     if (list == null) {
       return null;
     }
-    ImmutableSortedMap.Builder<String, Map<?, ?>> result =
-        new ImmutableSortedMap.Builder<>(String::compareTo);
-    list.forEach(value -> result.put(value, Collections.emptyMap()));
-    return result.build();
+    return list.stream()
+        .collect(
+            ImmutableSortedMap.toImmutableSortedMap(
+                String::compareTo, elementMapper, key -> Collections.emptyMap()));
```

Comment:
In general, for unused variables, we name them `ignored` to avoid any confusion - s/`key`/`ignored`/

Reply:
changed.